app/models/user.rb

```diff
@@ -40,9 +40,10 @@ module Permissions
     NONE  = 0
     ADMIN = 1
     INSTRUCTOR = 2
+    SUPER_ADMIN = 3
```

Comment:
Hmm... I wonder if we have any stray places that check against `user.permissions` directly rather than go through `user.admin?`.

You could do a preliminary check for this by (temporarily) changing the user factory's admin method to use a super admin, and see if it breaks any specs.

Reply:
I just checked this, the only tests that fail are:
 - the ones that ensure admins don't have permission for things reserved only for super-admins (duh)
 - one test in `models/alert_spec`. Backtrace below. I don't _think_ this is a problem but let me know. ```3) Alert sending emails still sends emails for other alert types
     Failure/Error: DEFAULT_FAILURE_NOTIFIER = lambda { |failure, _opts| raise failure }
       Exactly one instance should have received the following message(s) but didn't: alert
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-support-3.7.0/lib/rspec/support.rb:97:in `block in <module:Support>'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-support-3.7.0/lib/rspec/support.rb:106:in `notify_failure'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-mocks-3.7.0/lib/rspec/mocks/error_generator.rb:327:in `notify'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-mocks-3.7.0/lib/rspec/mocks/error_generator.rb:311:in `__raise'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-mocks-3.7.0/lib/rspec/mocks/any_instance/error_generator.rb:7:in `raise_second_instance_received_message_error'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-mocks-3.7.0/lib/rspec/mocks/any_instance/recorder.rb:95:in `verify'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-mocks-3.7.0/lib/rspec/mocks/space.rb:75:in `block in verify_all'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-mocks-3.7.0/lib/rspec/mocks/space.rb:75:in `each_value'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-mocks-3.7.0/lib/rspec/mocks/space.rb:75:in `verify_all'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-mocks-3.7.0/lib/rspec/mocks.rb:45:in `verify'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.0/lib/rspec/core/mocking_adapters/rspec.rb:23:in `verify_mocks_for_rspec'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.0/lib/rspec/core/example.rb:514:in `verify_mocks'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.0/lib/rspec/core/example.rb:508:in `run_after_example'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.0/lib/rspec/core/example.rb:273:in `block in run'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.0/lib/rspec/core/example.rb:500:in `block in with_around_and_singleton_context_hooks'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.0/lib/rspec/core/example.rb:457:in `block in with_around_example_hooks'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.0/lib/rspec/core/hooks.rb:466:in `block in run'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.0/lib/rspec/core/hooks.rb:606:in `block in run_around_example_hooks_for'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.0/lib/rspec/core/example.rb:342:in `call'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-rails-3.7.2/lib/rspec/rails/adapters.rb:127:in `block (2 levels) in <module:MinitestLifecycleAdapter>'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.0/lib/rspec/core/example.rb:447:in `instance_exec'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.0/lib/rspec/core/example.rb:447:in `instance_exec'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.0/lib/rspec/core/hooks.rb:375:in `execute_with'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.0/lib/rspec/core/hooks.rb:608:in `block (2 levels) in run_around_example_hooks_for'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.0/lib/rspec/core/example.rb:342:in `call'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.0/lib/rspec/core/hooks.rb:609:in `run_around_example_hooks_for'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.0/lib/rspec/core/hooks.rb:466:in `run'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.0/lib/rspec/core/example.rb:457:in `with_around_example_hooks'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.0/lib/rspec/core/example.rb:500:in `with_around_and_singleton_context_hooks'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.0/lib/rspec/core/example.rb:251:in `run'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.0/lib/rspec/core/example_group.rb:628:in `block in run_examples'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.0/lib/rspec/core/example_group.rb:624:in `map'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.0/lib/rspec/core/example_group.rb:624:in `run_examples'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.0/lib/rspec/core/example_group.rb:590:in `run'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.0/lib/rspec/core/example_group.rb:591:in `block in run'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.0/lib/rspec/core/example_group.rb:591:in `map'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.0/lib/rspec/core/example_group.rb:591:in `run'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.0/lib/rspec/core/runner.rb:118:in `block (3 levels) in run_specs'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.0/lib/rspec/core/runner.rb:118:in `map'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.0/lib/rspec/core/runner.rb:118:in `block (2 levels) in run_specs'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.0/lib/rspec/core/configuration.rb:1896:in `with_suite_hooks'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.0/lib/rspec/core/runner.rb:113:in `block in run_specs'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.0/lib/rspec/core/reporter.rb:79:in `report'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.0/lib/rspec/core/runner.rb:112:in `run_specs'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.0/lib/rspec/core/runner.rb:87:in `run'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.0/lib/rspec/core/runner.rb:71:in `run'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.0/lib/rspec/core/runner.rb:45:in `invoke'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/rspec-core-3.7.0/exe/rspec:4:in `<top (required)>'
     # /Users/jstoebel/.rbenv/versions/2.3.1/bin/rspec:23:in `load'
     # /Users/jstoebel/.rbenv/versions/2.3.1/bin/rspec:23:in `<top (required)>'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/bundler-1.16.1/lib/bundler/cli/exec.rb:75:in `load'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/bundler-1.16.1/lib/bundler/cli/exec.rb:75:in `kernel_load'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/bundler-1.16.1/lib/bundler/cli/exec.rb:28:in `run'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/bundler-1.16.1/lib/bundler/cli.rb:424:in `exec'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/bundler-1.16.1/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/bundler-1.16.1/lib/bundler/vendor/thor/lib/thor/invocation.rb:126:in `invoke_command'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/bundler-1.16.1/lib/bundler/vendor/thor/lib/thor.rb:387:in `dispatch'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/bundler-1.16.1/lib/bundler/cli.rb:27:in `dispatch'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/bundler-1.16.1/lib/bundler/vendor/thor/lib/thor/base.rb:466:in `start'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/bundler-1.16.1/lib/bundler/cli.rb:18:in `start'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/bundler-1.16.1/exe/bundle:30:in `block in <top (required)>'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/bundler-1.16.1/lib/bundler/friendly_errors.rb:122:in `with_friendly_errors'
     # /Users/jstoebel/.rbenv/versions/2.3.1/lib/ruby/gems/2.3.0/gems/bundler-1.16.1/exe/bundle:22:in `<top (required)>'
     # /Users/jstoebel/.rbenv/versions/2.3.1/bin/bundle:23:in `load'
     # /Users/jstoebel/.rbenv/versions/2.3.1/bin/bundle:23:in `<main>'
     #
     #   Showing full backtrace because every line was filtered out.
     #   See docs for RSpec::Configuration#backtrace_exclusion_patterns and
     #   RSpec::Configuration#backtrace_inclusion_patterns for more information.```